source/deployment/cluster.rst

```diff
@@ -0,0 +1,206 @@
+
+Manual failover for master database  
+`````````````````````````````````````````````````````````````````
+If the need arises to switch from the current master database--for example, if it is running out of disk space, or requires maintenance updates, or for other reasons--the Mattermost server can switch to using one of its read replicas as a master database by updating ``DataSource`` in ``config.json``. The following procedure can be used to apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console** > **Configuration** and press **Reload Configuration from Disk** to reload configuration settings for the Mattermost server from ``config.json``. 
+2. Go to **System Console** > **Database** and press **Recycle Database Connections** to takedown existing database connections and set up new connections in the multi-database configuration. 
+
+While connection settings are changing there may be a brief moment when writes to the master database will be unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+
+Transparent Failover
+````````````````````````````````````
+The database can be configured for high availability and transparent failover utilizing the existing database technologies.  We recommend MySQL Clustering, Postgres Clustering, or Amazon Aoura.  Database transparent failover is beyond the scope of this documentation.
+
+-----
+
+Troubleshooting
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Red Server Status
+---------------------------
+When high availability is enabled, the System Console displays the server status as red or green, indicating if the servers are communicating correctly with the cluster. The servers use inter-node communication to ping the other machines in the cluster, and once a ping is established the servers exchange information, such as server version and configuration files. Red server status may display for the following reasons:
+
+- **Configuration file mismatch between the servers**: Mattermost will still attempt the inter-node communication, but the System Console will show a red status for the since the high availability feature assumes the same server version and configuration file to function properly.
+- **Server is down**: If an inter-node communication fails to send a message it will attempt again in 15 seconds.  If the second attempt fails, the server is assumed to be down. An error message is written to the logs and the System Console will show a status of red for that server.
```

Comment:
Maybe added 1 more sentence. 

`The other nodes will attempt to ping and establish a connection with the down server every 15 seconds until it rejoins the cluster.`